src/basic/pidref.c

```diff
@@ -27,14 +48,17 @@ bool pidref_equal(const PidRef *a, const PidRef *b) {
                 if (a->fd < 0 || b->fd < 0)
                         return true;
 
-                /* pidfds live in their own pidfs and each process comes with a unique inode number since
-                 * kernel 6.8. We can safely do this on older kernels too though, as previously anonymous
-                 * inode was used and inode number was the same for all pidfds. */
-                r = fd_inode_same(a->fd, b->fd);
-                if (r < 0)
-                        log_debug_errno(r, "Failed to check whether pidfds for pid " PID_FMT " are equal, assuming yes: %m",
-                                        a->pid);
-                return r != 0;
+                PidRef *p;
+                FOREACH_ARGUMENT(p, a, b) {
+                        r = pidref_acquire_pidfd_id(p);
```

Comment:
i'd really prefer if we acquire the ID right-away when we initialize the pidref, and not delay it.

yes, this might mean we do a syscall more initially, but i think that should be ok

Reply:
So my idea is that we should be able to construct/copy pidrefs freely, and that assumption would break if we now assume the ino is available right away. Also, `pidref_equal` is not called that frequently, and thus there's no need to `fstat()` each pidfd we acquired.

Reply:
E.g. 707277716 changed our serialization logic to send both pid and pidfd, and construct a pidref locally on deserialization. And I don't want to needlessly modify that.

Reply:
i think we should start to serialize this new id too